core/src/main/java/com/scalar/db/storage/jdbc/JdbcTableMetadataManager.java

```
@@ -105,26 +113,291 @@ public JdbcTableMetadataManager(
+      int ordinalPosition,
+      String column)
+      throws SQLException {
+    KeyType keyType = null;
+    if (metadata.getPartitionKeyNames().contains(column)) {
+      keyType = KeyType.PARTITION;
+    }
+    if (metadata.getClusteringKeyNames().contains(column)) {
+      keyType = KeyType.CLUSTERING;
+    }
+
+    try {
+      String insertStatement =
+          getInsertStatement(
+              schema,
+              table,
+              column,
+              metadata.getColumnDataType(column),
+              keyType,
+              metadata.getClusteringOrder(column),
+              metadata.getSecondaryIndexNames().contains(column),
+              ordinalPosition);
+      execute(connection, insertStatement);
+    } catch (SQLException e) {
+      connection.rollback();
+      throw e;
+    }
+  }
+
+  private void createTableIfNotExists(Connection connection) throws SQLException {
+    String createTableStatement =
+        String.format("CREATE TABLE %s(", encloseFullTableNames(getMetadataSchema(), TABLE))
+            + String.format("%s %s,", enclose(FULL_TABLE_NAME), getTextType(128))
+            + String.format("%s %s,", enclose(COLUMN_NAME), getTextType(128))
+            + String.format("%s %s NOT NULL,", enclose(DATA_TYPE), getTextType(20))
+            + String.format("%s %s,", enclose(KEY_TYPE), getTextType(20))
+            + String.format("%s %s,", enclose(CLUSTERING_ORDER), getTextType(10))
+            + String.format("%s %s NOT NULL,", enclose(INDEXED), getBooleanType())
+            + String.format("%s INTEGER NOT NULL,", enclose(ORDINAL_POSITION))
+            + String.format(
+                "PRIMARY KEY (%s, %s))", enclose(FULL_TABLE_NAME), enclose(COLUMN_NAME));
+    String createTableIfNotExistsStatement;
+    switch (rdbEngine) {
+      case ORACLE:
+        // Save into the tableExistCount variable if the metadata table
+        // exists (1=true, 0=false) by querying a system table
+        String selectTableExistStatement =
```

I'm curious what would happen if you create a table that exists in Oracle and SQLServer.
If it's throwing a certain exception for duplicate table creation,  I think simply ignoring the exception might be better rather than select and create in a stored procedure.
I'm sure you have tried it and came to this but asking just in case.